working-with-collections/main.js

```diff
@@ -0,0 +1,110 @@
+/* global _ */
+/* eslint-disable no-console */
+console.log('Lodash is loaded:', typeof _ !== 'undefined');
+var deck = [];
+var scores = [];
+var winningScore;
+var congrats = '';
+var players = [
+  {
+    name: 'Rachel',
+    hand: [],
+    score: 0
+  },
+  {
+    name: 'Stephen',
+    hand: [],
+    score: 0
+  },
+  {
+    name: 'Ruby',
+    hand: [],
+    score: 0
+  },
+  {
+    name: 'Kris',
+    hand: [],
+    score: 0
+  }
+];
+
+for (var i = 1; i <= 13; i++) {
+  var hearts = {};
+  hearts.rank = i;
+  hearts.suit = 'hearts';
+  hearts.value = i;
+  var spades = {};
+  spades.rank = i;
+  spades.suit = 'spades';
+  spades.value = i;
+  var diamonds = {};
+  diamonds.rank = i;
+  diamonds.suit = 'diamonds';
+  diamonds.value = i;
+  var clubs = {};
+  clubs.rank = i;
+  clubs.suit = 'clubs';
+  clubs.value = i;
+  deck.push(hearts);
+  deck.push(spades);
+  deck.push(diamonds);
+  deck.push(clubs);
+  updateDeck();
+}
```

Comment:
This could be handled more elegantly by creating an array of ranks and an array of suits. A nested `for` loop can be used to generate the deck.

```js
var ranks = [2, 3, 4, 5, 6, 7, 8, 9, 10, 'jack', 'queen', 'king'];
var suits = ['spades', 'diamonds', 'clubs', 'hearts'];
```